core/modules/ccontrol/UserQueryFactory.cc

```diff
@@ -194,3 +240,41 @@ UserQueryFactory::Impl::Impl(StringMap const& m) {
 }
 
 }}} // lsst::qserv::ccontrol
+
+
+namespace {
+
+// regex for DROP TABLE [dbname.]table; both table and db names can be in quotes;
+// db name will be in group 3, table name in group 5.
+// Note that parens around whole string are not part of the regex but raw string literal
+std::regex _dropTableRe(R"(^drop\s+table\s+((["`]?)(\w+)\2[.])?(["`]?)(\w+)\4\s*;?\s*$)",
+                        std::regex::ECMAScript | std::regex::icase | std::regex::optimize);
+
+bool isDropTable(std::string const& query, std::string& dbName, std::string& tableName) {
+    LOGF(_log, LOG_LVL_DEBUG, "isDropTable: %s" % query);
+    std::smatch sm;
+    if (regex_match(query, sm, _dropTableRe)) {
+        dbName = sm.str(3);
+        tableName = sm.str(5);
+        LOGF(_log, LOG_LVL_DEBUG, "isDropTable: match: %s.%s" % dbName % tableName);
+        return true;
+    }
+    return false;
+}
+
+// regex for SELECT *
+// Note that parens around whole string are not part of the regex but raw string literal
+std::regex _selectRe(R"(^select\s+.+$)",
+                     std::regex::ECMAScript | std::regex::icase | std::regex::optimize);
+
+bool isSelect(std::string const& query) {
+    LOGF(_log, LOG_LVL_DEBUG, "isSelect: %s" % query);
+    std::smatch sm;
+    bool match = regex_match(query, sm, _selectRe);
+    if (match) {
```

Comment:
Well IIRC it was `return regex_match(query, sm, _selectRe);` and then I wanted to add log message and ended up with this code. It can be done the same way above, ti also returns the same value as the one returned from `regex_match()`. I'll change code above to make it look similar.